### ISSUE TITLE
regression: trait bound is not satisfied Cow/Borrow<str>

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#127339]
Not clear what exactly has changed.

* https://crater-reports.s3.amazonaws.com/beta-1.80-4/beta-2024-06-22/reg/blades-0.5.0/log.txt

```
[INFO] [stdout] error[E0277]: the trait bound `for<'a> beef::generic::Cow<'a, str, beef::wide::internal::Wide>: Borrow<beef::generic::Cow<'_, str, lean::internal::Lean>>` is not satisfied
[INFO] [stdout]    --> src/page.rs:433:18
[INFO] [stdout]     |
[INFO] [stdout] 433 |             .get(template)
[INFO] [stdout]     |              --- ^^^^^^^^ the trait `for<'a> Borrow<beef::generic::Cow<'_, str, lean::internal::Lean>>` is not implemented for `beef::generic::Cow<'a, str, beef::wide::internal::Wide>`
[INFO] [stdout]     |              |
[INFO] [stdout]     |              required by a bound introduced by this call
[INFO] [stdout]     |
[INFO] [stdout]     = help: the trait `Borrow<str>` is implemented for `beef::generic::Cow<'_, str, beef::wide::internal::Wide>`
[INFO] [stdout]     = help: for that trait implementation, expected `str`, found `beef::generic::Cow<'_, str, lean::internal::Lean>`
[INFO] [stdout] note: required by a bound in `Ramhorns::<H>::get`
[INFO] [stdout]    --> /opt/rustwide/cargo-home/registry/src/index.crates.io-6f17d22bba15001f/ramhorns-0.14.0/src/lib.rs:206:31
[INFO] [stdout]     |
[INFO] [stdout] 204 |     pub fn get<S>(&self, name: &S) -> Option<&Template<'static>>
[INFO] [stdout]     |            --- required by a bound in this associated function
[INFO] [stdout] 205 |     where
[INFO] [stdout] 206 |         for<'a> Cow<'a, str>: std::borrow::Borrow<S>,
[INFO] [stdout]     |                               ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Ramhorns::<H>::get`
```

**Comment by compiler-errors** [comment]
This is https://github.com/rust-lang/rust/pull/119820, cc @lcnr (though we probably already have this listed as a regression in that one)

**Comment by apiraino** [comment]
From me, assigning priority with same reasoning as for #127342 

WG-prioritization assigning priority ([Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.23127339.20regression.3A.20trait.20bound.20is.20not.20satisfied.20Cow.2FBorrow.E2.80.A6)).

@rustbot label -I-prioritize +P-high

**Comment by apiraino** [comment]
IIUC this is now fixed by #127568 (now beta-backported).

If yes, I think this issue can be closed
